skycoin/skywire
# Adapt route encryption for old visors

Did you run `make format && make check`?
Yes
Fixes #501 

 Changes:	
- Handshake packet type was added;
- Visor now exchange handshake packets while creating route groups. Initiator sends it first, remote visor responds. There are 3 possible cases here:
1) both visors support route encryption. in this case handshake packets are exchanged, visors start communicating with the traffic encrypted;
2) initiator supports encryption, while remote visor is not. in this case initiator sends handshake and waits for response. either it gets no response during timeout interval (2 seconds now) or it gets data packet. in both cases remote visor is known to not support the encryption, communication goes without it;
3) initiator doesn't support encryption, while remote visor does. in this case remote visor waits for the handshake packet to come in. again, either 2 seconds timeout passes or data packet arrives. in both cases initiator is known to not support the encryption, communication goes without it.

How to test this PR:
1. Setup 2 visors with some apps (VPN server and client, for example).
2. Try to communicate. Should work for all visors supporting encryption and not.
